plugins/modules/mso_backup.py

```diff
@@ -180,11 +196,41 @@ def main():
                 mso.existing = mso.request('backups/backupRecords/{id}'.format(id=mso.existing[0].get('id')), method='DELETE')
         mso.exit_json()
 
-    path = 'backups'
+    if state == 'restore':
+        mso.previous = mso.existing
+        if len(mso.existing) > 1:
+            mso.module.fail_json(msg="Multiple backups with same name found. Existing backups with similar names: {0}".format(', '.join(backup_names)))
+        if module.check_mode:
+            mso.existing = mso.proposed
+        else:
+            mso.existing = mso.request('backups/{id}/restore'.format(id=mso.existing[0].get('id')), method='PUT')
+        mso.exit_json()
 
-    if state == 'present':
+    if state == 'download':
         mso.previous = mso.existing
+        payload = dict(destination=destination)
+        if len(mso.existing) > 1:
+            mso.module.fail_json(msg="Multiple backups with same name found. Existing backups with similar names: {0}".format(', '.join(backup_names)))
+        if module.check_mode:
+            mso.existing = mso.proposed
+        else:
+            mso.existing = mso.request('backups/{id}/download'.format(id=mso.existing[0].get('id')), method='GET', data=payload)
```

Comment:
I think you need to check that mso.existing length == 1 because mso.existing[0] will fail if not.